src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -909,21 +972,53 @@ private byte[] GetAsciiEncoding(byte value)
             return encoding;
         }
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private int DoesAsciiNeedEncoding(byte value)
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        private unsafe void InitializeAsciiCache()
         {
-            Debug.Assert(value <= 0x7F);
+            Debug.Assert(!_isAsciiCacheInitialized);
 
-            int needsEscaping = _asciiNeedsEscaping[value];
-
-            Debug.Assert(needsEscaping == 0 || needsEscaping == 1 || needsEscaping == -1);
+            for (int i = 0; i < _asciiNeedsEscaping.Length; i++)
+            {
+                _asciiNeedsEscaping[i] = WillEncode(i);
+            }
 
-            if (needsEscaping == 0)
+#if NETCOREAPP
+            if (Ssse3.IsSupported)
             {
-                needsEscaping = WillEncode(value) ? 1 : -1;
-                _asciiNeedsEscaping[value] = needsEscaping;
+                Debug.Assert(_asciiNeedsEscaping.Length == 0x80);
+
+                sbyte* tmp = stackalloc sbyte[Vector128<sbyte>.Count];
+
+                for (int lowNibble = 0; lowNibble <= 0xF; lowNibble++)
+                {
+                    int bitMask = 0;
+
+                    for (int highNibble = 0; highNibble <= 0x7; highNibble++)
+                    {
+                        int value = highNibble << 4 | lowNibble;
+
+                        if (WillEncode(value))
```

Comment:
I'll change this to reuse `_asciiNeedsEscaping` as it is already initialized, so it safes 128 calls to `WillEncode`.